quickwit/quickwit-indexing/src/source/kafka_source.rs

```diff
@@ -595,6 +626,37 @@ fn spawn_consumer_poll_loop(
     })
 }
 
+/// Attempts to drain the overflow buffer by sending as many events as possible to the channel
+/// `events_tx` before `max_duration` has passed.
+fn drain_overflow_buffer<T>(
+    overflow_buffer: &mut VecDeque<T>,
+    events_tx: &mpsc::Sender<T>,
+    max_duration: Duration,
+) {
+    if overflow_buffer.is_empty() {
+        return;
+    }
+    let now = tokio::time::Instant::now();
+    let deadline = now + max_duration;
+    Handle::current().block_on(async {
+        while let Some(event) = overflow_buffer.pop_back() {
+            let permit = match tokio::time::timeout_at(deadline, events_tx.reserve()).await {
```

Comment:
I don't think permit is required.

These channels are so called cancellable.
sending directly is still correct.

In other words, if the timeout error is returned, you have the guarantee that the message was not sent.